cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -535,6 +541,166 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, specification);
   }
 
+  @PUT
```

Comment:
This may be true in the pure REST philosophy. But we also want to be consistent with existing REST APIs such as PUT for creating a dataset, an app, or a stream.